### PR TITLE
docs(es): add Spanish translations

### LINKED ISSUE #3850: docs: add spanish translation
Hi guys, I hope you are well. I have forked to check spanish #translation. It is working for some files I have tested. I would like to contribute. Thanks.

### PR DESCRIPTION
**Pull request by caparra92**
closes #3850

I'm adding @JohnFreddyTutistar to help in the translation.

### CODE REVIEW

**Comment by caparra92** [comment]
Files in docs/es/guide translated to spanish.
@JohnFreddyTutistar has added docs/es/reference in [https://github.com/vuejs/vitepress/pull/3934](url)

Please let me know if anything else is needed.

File: docs/.vitepress/config/index.ts [hunk 1]
```diff
@@ -12,6 +13,7 @@ export default defineConfig({
     zh: { label: '简体中文', ...zh },
     pt: { label: 'Português', ...pt },
     ru: { label: 'Русский', ...ru },
-    ko: { label: '한국어', lang: 'ko-KR', link: 'https://vitepress.vuejs.kr/' }
+    ko: { label: '한국어', lang: 'ko-KR', link: 'https://vitepress.vuejs.kr/' },
+    es: { label: 'Español', ...es }
```
Comment (brc-dd) [inline docs/.vitepress/config/index.ts:17]: let's keep built in ones before external translations

```suggestion
    es: { label: 'Español', ...es }.
    ko: { label: '한국어', lang: 'ko-KR', link: 'https://vitepress.vuejs.kr/' }
```

**Review by brc-dd — APPROVED** [review]

**Comment by caparra92** [comment]
There are a few typos I like to fix. Can I use this PR or it is necessary to open a new one? Thanks.

**Comment by brc-dd** [comment]
You’ll need to open a new one.

**Comment by caparra92** [comment]
@JohnFreddyTutistar is going to open a new one.

**Pull request merged**